### PR TITLE
GTK3 Support [v2]

### PR DESCRIPTION
**Pull request by bp0**
This PR replaces #113 (and #59). It corrects the Author on @TotalCaesar659's work and squashes all of my commits into one.

To test:
```
mkdir build && cd build
cmake -DHARDINFO_GTK3=1 ..
[etc.]
```

Hardinfo builds and runs without crashing on GTK3.
* there are many deprecation warnings during build.
* loadgraph is replaced by Christian Hergert's uber-graph widget for GTK3 hardinfo
* guibench is disabled for gtk3.
* gtk2 continues to work as well as it ever did.
* look for TODO:GTK3 in the code to find places that need work.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -176,31 +193,70 @@ foreach (_module ${HARDINFO_MODULES})
...
 
-add_library(hardinfo-shell STATIC
-        shell/callbacks.c
-        shell/iconcache.c
-        shell/loadgraph.c
-        shell/menu.c
-        shell/report.c
-        shell/shell.c
-        shell/stock.c
+if (HARDINFO_GTK3)
```
Comment (lpereira) [inline CMakeLists.txt]: Can uber-graph be used in GTK2? If so, we can just get rid of the old loadgraph code and use it instead.

This should simplify the build system as well by removing some conditionals and copied lists of files.

File: hardinfo/util.c [hunk 2]
```diff
@@ -979,6 +994,9 @@ gint tree_view_get_visible_height(GtkTreeView * tv)
 
 void tree_view_save_image(gchar * filename)
```
Comment (lpereira) [inline hardinfo/util.c]: This whole function can be removed, it's not used for anything. (It's a leftover from a feature I've never finished.)

File: includes/loadgraph.h [hunk 3]
```diff
@@ -39,37 +33,20 @@ typedef enum {
...
-    gint	   width, height;
-    LoadGraphColor color;    
-    
-    gint	   max_value, remax_count;
-    
-    PangoLayout   *layout;
-    gchar	  *suffix;
-};
+void         load_graph_update(LoadGraph *lg, gdouble value);
+void         load_graph_update_ex(LoadGraph *lg, guint line, gdouble value);
```
Comment (lpereira) [inline includes/loadgraph.h]: There are very few places where `load_graph_update()` is called. Maybe it would be better to make `load_graph_update_ex()` the default and only behavior to update the load graph.

File: shell/callbacks.c [hunk 4]
```diff
@@ -44,13 +44,23 @@ void cb_save_graphic()
     gchar *filename;
```
Comment (lpereira) [inline shell/callbacks.c]: The `cb_save_graphic()` function can be removed as well, nothing calls it any longer.

File: shell/loadgraph.c [hunk 5]
```diff
@@ -20,11 +20,38 @@
...
  * License along with the Simple Load Graph; if not, write to the Free
  * Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
- * 02111-1307 USA. 
+ * 02111-1307 USA.
  */
 
 #include "loadgraph.h"
 
+struct _LoadGraph {
+#if GTK_CHECK_VERSION(3, 0, 0)
```
Comment (lpereira) [inline shell/loadgraph.c]: Is this change necessary if uber-graph is being used for GTK3 now?

File: shell/report.c [hunk 6]
```diff
@@ -636,15 +646,21 @@ static gboolean report_generate(ReportDialog * rd)
 					GTK_MESSAGE_QUESTION,
 					GTK_BUTTONS_NONE,
 					_("Open the report with your web browser?"));
+#if GTK_CHECK_VERSION(3, 0, 0)
+    gtk_dialog_add_buttons(GTK_DIALOG(dialog),
+			       "_No", GTK_RESPONSE_REJECT,
```
Comment (lpereira) [inline shell/report.c]: Shouldn't these be translatable?

File: shell/shell.c [hunk 7]
```diff
@@ -33,6 +33,10 @@
 
 #include "callbacks.h"
 
+#ifndef gtk_notebook_set_page
```
Comment (lpereira) [inline shell/shell.c]: Better to use the `GTK_CHECK_VERSION` macro here.

File: CMakeLists.txt (hunk 1 above)
Comment (bp0) [inline CMakeLists.txt]: uber-graph requires gtk3.

**Comment by lpereira** [comment]
Merged, thanks @bp0 and @TotalCaesar659!
